src/hooks/index.ts

```diff
@@ -116,6 +116,15 @@ type Web3React = {
 
 export function useWeb3React(): Web3React {
   const { connector, chainId, account, isActive: active, provider } = useWeb3ReactCore<Web3Provider>()
+
+  useEffect(() => {
+    provider?.getNetwork().then(res => {
+      if (res.chainId !== chainId) {
+        window.location.reload()
+      }
+    })
+  }, [chainId, provider])
```

Comment:
This hook should move to somewhere only be instance only once like `src/pages/App.tsx`

Reply:
this is a workaround solution, I will dig into web3-react lib later. 
btw, this is first place retrieve chainId, that why I place this hook here. move to App will be hard to manage